osbuild/util/linux.py

```diff
@@ -0,0 +1,139 @@
+"""Linux API Access
+
+This module provides access to linux system-calls and other APIs, in particular
+those not provided by the python standard library. The idea is to provide
+universal wrappers with broad access to linux APIs. Convenience helpers and
+higher-level abstractions are beyond the scope of this module.
+
+In some cases it is overly complex to provide universal access to a specifc
+API. Hence, the API might be restricted to a reduced subset of its
+functionality, just to make sure we can actually implement the wrappers in a
+reasonable manner.
+"""
+
+
+import fcntl
+import os
+import struct
+
+
+__all__ = [
+    "fcntl_flock",
+]
+
+
+def fcntl_flock(fd, lock_type):
+    """Perform File-locking Operation
+
+    This function performs a linux file-locking operation on the specified
+    file-descriptor. The specific type of lock must be specified by the caller.
+    This function does not allow to specify the byte-range of the file to lock.
+    Instead, it always applies the lock operations to the entire file.
+
+    For system-level documentation, see the `fcntl(2)` man-page, especially the
+    section about `struct flock` and the locking commands.
+
+    This function always uses the open-file-description locks provided by
+    modern linux kernels. This means, locks are tied to the
+    open-file-description. That is, they are shared between duplicated
+    file-descriptors. Furthermore, acquiring a lock while already holding a
+    lock will update the lock to the new specified lock type, rather than
+    acquiring a new lock.
+
+    So far, this function always performs try-lock operations. The blocking
+    mode is currently not exposed.
+
+    Parameters
+    ----------
+    fd : int
```

Comment:
What about having those as actual type annotations on the parameters, which can be picked up by IDEs/language servers.

Reply:
Right, that sounds like the correct thing to do. Then we can drop the annotations here.